src/llvm-version.h

```diff
@@ -2,53 +2,37 @@
 
 #include <llvm/Config/llvm-config.h>
 
-#if defined(LLVM_VERSION_MAJOR) && LLVM_VERSION_MAJOR == 4 && LLVM_VERSION_MINOR >= 0
-#define LLVM40 1
-#endif
-
-#if defined(LLVM40) || (defined(LLVM_VERSION_MAJOR) && LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 9)
-#define LLVM39 1
-#endif
-
-#if defined(LLVM40) || (defined(LLVM_VERSION_MAJOR) && LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 8)
-#define LLVM38 1
-#define USE_ORCJIT
-#endif
-
-#if defined(LLVM40) || (defined(LLVM_VERSION_MAJOR) && LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 7)
-#define LLVM37 1
+// The version LLVM version used, LLVM_VERSION, is represented as a hexadecimal
+// string: 0xABBCC where A is the major version, B is minor, and C is patch. So
+// for example, LLVM 3.7.0 is 0x30700.
+#ifdef LLVM_VERSION_MAJOR
 
-// We enable ORCJIT only if we have our custom patches
-#ifndef SYSTEM_LLVM
-#define USE_ORCJIT
-#endif
+    #define LLVM_VERSION (LLVM_VERSION_MAJOR * 0x10000 + LLVM_VERSION_MINOR * 0x100 \
```

Comment:
And I think we should use a `JL_` prefix. This name is too close to what LLVM might decide to define some day.